janusgraph-core/src/main/java/org/janusgraph/graphdb/query/index/AbstractIndexSelectionStrategy.java

```diff
@@ -215,7 +216,12 @@ private boolean coversAll(final MixedIndexType index, Condition<JanusGraphElemen
             .filter(field -> field.getStatus() == SchemaStatus.ENABLED)
             .filter(field -> field.getFieldKey().equals(key))
             .findAny().orElse(null);
-        return match != null && indexInfo.supports(index, match, atom.getPredicate());
+        if (match == null) {
+            return false;
+        }
+        // Although Full-text search does not support NOT_EQUAL query, NOT_EQUAL(null) is an exception since it matches all
+        boolean isAllMatch = atom.getPredicate() == Cmp.NOT_EQUAL && atom.getValue() == null;
+        return isAllMatch || indexInfo.supports(index, match, atom.getPredicate());
```

Comment:
This uncovers an existing issue (https://github.com/JanusGraph/janusgraph/issues/2176), which causes CI to fail. If we drop this part, OR fix #2176, OR comment out the flaky assertion, the CI should pass.

Reply:
This PR is based on PR https://github.com/JanusGraph/janusgraph/pull/2182